### ISSUE TITLE
Read `MultibandTile`s according to `LayoutDefinition` directly from GeoTiff

### ISSUE THREAD

**Issue by echeipesh** [locationtech/geotrellis#2700]
Connects: https://github.com/locationtech/geotrellis/issues/2698

The main feature here is to impose a grid on source rasters before they are read. Mostly this avoids having to use arbitrary grid for IO and then using a spark shuffle to re-grid the files into intended layout.

1. List all GeoTiff files in a `file://` URI
2. Read raster metadata into case classes that describe the files
3. Split each `RasterRef` into window keyed by a `LayoutDefinition`
4. Partition resulting records such that read produces balanced RDD
5. Read windows in parallel

**Comment by metasim** [comment]
For point of reference, this is how we're dealing with this in RasterFrames, at the per-GeoTiff level:

https://github.com/locationtech/rasterframes/blob/593e45653f2d966a79e3dc3a36572ada5d02b7af/datasource/src/main/scala/astraea/spark/rasterframes/datasource/geotiff/GeoTiffRelation.scala#L88-L102

I /think/ 1 & 2 are covered... is 2 just the TLM?

Another issue we've not figured out is basically having to copy the TLM in each row. It's not a huge amount of data, but it's not exactly small either, and not friendly to joins. Maybe we need a separate unique identifier for referencing the TLM under which it lives.

What's a `RasterRef`?
